components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/authz/AuthorizationHandlerManager.java

```diff
@@ -229,6 +433,16 @@ private void addAuthorizedInternalScopes(OAuthAuthzReqMessageContext authzReqMsg
         authzReqMsgCtx.setApprovedScope(scopesToReturn);
     }
 
+    private void addRequestedOIDCScopes(OAuthAuthzReqMessageContext authzReqMsgCtx,
+                                             String[] requestedOIDCScopes) {
+        Set<String> scopesToReturn = new HashSet<>(Arrays.asList(authzReqMsgCtx.getApprovedScope()));
```

Comment:
Are we using a set here to remove duplicates?

Reply:
yes